libraries/Zigbee/src/ep/ZigbeeTemperatureDimmableLight.cpp

```diff
@@ -0,0 +1,178 @@
+    if (message->attribute.id == ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMPERATURE_ID && message->attribute.data.type == ESP_ZB_ZCL_ATTR_TYPE_U16) {
+      if (_current_color_temperature != *(uint16_t *)message->attribute.data.value) {
+        _current_color_temperature = *(uint16_t *)message->attribute.data.value;
+        lightChanged();
+      }
+      return;
+
+    } else {
+      log_w("Received message ignored. Attribute ID: %d not supported for Temperature Dimmable Light", message->attribute.id);
+    }
+  } else {
+    log_w("Received message ignored. Cluster ID: %d not supported for Temperature Dimmable Light", message->info.cluster);
+  }
+}
+
+void ZigbeeTemperatureDimmableLight::lightChanged() {
+  if (_on_light_change) {
+    _on_light_change(_current_state, _current_level, _current_color_temperature);
+  }
+}
+
+bool ZigbeeTemperatureDimmableLight::setLight(bool state, uint8_t level, uint16_t mireds) {
+  esp_zb_zcl_status_t ret = ESP_ZB_ZCL_STATUS_SUCCESS;
+  _current_state = state;
+  _current_level = level;
+  _current_color_temperature = mireds;
+  lightChanged();
+
+  log_v("Updating light state: %d, level: %d, color temperature: %d", state, level, mireds);
+  /* Update light clusters */
+  esp_zb_lock_acquire(portMAX_DELAY);
+  //set on/off state
+  ret = esp_zb_zcl_set_attribute_val(
+    _endpoint, ESP_ZB_ZCL_CLUSTER_ID_ON_OFF, ESP_ZB_ZCL_CLUSTER_SERVER_ROLE, ESP_ZB_ZCL_ATTR_ON_OFF_ON_OFF_ID, &_current_state, false
+  );
+  if (ret != ESP_ZB_ZCL_STATUS_SUCCESS) {
+    log_e("Failed to set light state: 0x%x: %s", ret, esp_zb_zcl_status_to_name(ret));
+    goto unlock_and_return;
+  }
+  //set level
+  ret = esp_zb_zcl_set_attribute_val(
+    _endpoint, ESP_ZB_ZCL_CLUSTER_ID_LEVEL_CONTROL, ESP_ZB_ZCL_CLUSTER_SERVER_ROLE, ESP_ZB_ZCL_ATTR_LEVEL_CONTROL_CURRENT_LEVEL_ID, &_current_level, false
+  );
+  if (ret != ESP_ZB_ZCL_STATUS_SUCCESS) {
+    log_e("Failed to set light level: 0x%x: %s", ret, esp_zb_zcl_status_to_name(ret));
+    goto unlock_and_return;
+  }
+  //set color temperature
+  ret = esp_zb_zcl_set_attribute_val(
+    _endpoint, ESP_ZB_ZCL_CLUSTER_ID_COLOR_CONTROL, ESP_ZB_ZCL_CLUSTER_SERVER_ROLE, ESP_ZB_ZCL_ATTR_COLOR_CONTROL_COLOR_TEMPERATURE_ID,
+    &_current_color_temperature, false
+  );
+  if (ret != ESP_ZB_ZCL_STATUS_SUCCESS) {
+    log_e("Failed to set light color temperature: 0x%x: %s", ret, esp_zb_zcl_status_to_name(ret));
+    goto unlock_and_return;
+  }
+unlock_and_return:
+  esp_zb_lock_release();
+  return ret == ESP_ZB_ZCL_STATUS_SUCCESS;
+}
+
+bool ZigbeeTemperatureDimmableLight::setLightState(bool state) {
+  return setLight(state, _current_level, _current_color_temperature);
+}
+
+bool ZigbeeTemperatureDimmableLight::setLightLevel(uint8_t level) {
+  return setLight(_current_state, level, _current_color_temperature);
+}
+
+bool ZigbeeTemperatureDimmableLight::setColorTemperature(uint16_t mireds) {
+  return setLight(_current_state, _current_level, mireds);
+}
```

Comment:
I would prefer the setLightTemperature method name instead.
```suggestion
bool ZigbeeTemperatureDimmableLight::setLightTemperature(uint16_t mireds) {
  return setLight(_current_state, _current_level, mireds);
}
```